lib/auth0/exception.rb

```diff
@@ -25,12 +41,18 @@ class MissingClientId < Auth0::Exception; end
   # exception for an unset parameter
   class MissingParameter < Auth0::Exception; end
   # Api v2 access denied
-  class AccessDenied < Auth0::Exception; end
+  class AccessDenied < Auth0::HTTPError; end
   # Invalid parameter passed, e.g. empty where ID is required
   class InvalidParameter < Auth0::Exception; end
   # Invalid Auth0 credentials either client_id/secret for API v1
   # or JWT for API v2/
   class InvalidCredentials < Auth0::Exception; end
   # Invalid Auth0 API namespace
   class InvalidApiNamespace < Auth0::Exception; end
+  # Auth0 API rate-limiting encountered
+  class RateLimitEncountered < Auth0::HTTPError
+    def reset
+      Time.at(headers['X-RateLimit-Reset']).utc
+    end
+  end
```

Comment:
Do you think it would be helpful to have a `limit` method here as well?

Reply:
I couldn't think of a use-case for it, given that you won't find out until after you've already hit the limit; but it would be pretty easy to include, if we want it.